__test__/firebase/ManageAnnouncements.test.ts

```diff
@@ -0,0 +1,89 @@
+import { Firestore } from "firebase/firestore"
+import { showErrorToast, showSuccessToast } from "../../components/ToastMessage/toast"
+import { createAnnouncement, getAllAnnouncements } from "../../firebase/ManageAnnouncements"
+import { waitFor } from "@testing-library/react-native"
+
+
+jest.mock("../../firebase/firebaseConfig", () => ({
+    db: jest.fn(() => ({} as Firestore))
+  }))
+  
+  const mockGetDoc = jest.fn().mockImplementation((uid: string) =>{
+    if (uid === "0") {
+      return Promise.resolve({ exists: jest.fn(() => false)})
+    }
+    else if (uid === "1") {
+      return Promise.resolve({ exists: jest.fn(() => true)})
+    }
+    else {
+      return Promise.reject()
+    }
```

Comment:
Could you test the case where we try to create an uid similar to another. This should reject the creation, I think that's an important test as it caused a lot of problems before.